diff -Npur a/data/gnome-shell/pamac-updates@manjaro.org/extension.js b/data/gnome-shell/pamac-updates@manjaro.org/extension.js
--- a/data/gnome-shell/pamac-updates@manjaro.org/extension.js	2026-10-08 08:54:42.906778117 -0600
+++ b/data/gnome-shell/pamac-updates@manjaro.org/extension.js	2026-10-08 09:09:52.403123694 -0600
@@ -80,7 +80,7 @@ class PamacUpdateIndicator extends Butto
 
 		this.updateIcon = new St.Icon({icon_name: "pamac-tray-no-update", style_class: 'system-status-icon'});
 
-		let box = new St.BoxLayout({ vertical: false, style_class: 'panel-status-menu-box' });
+		let box = new St.BoxLayout({ orientation: Clutter.Orientation.VERTICAL = 0, style_class: 'panel-status-menu-box' });
 
 		box.add_child(this.updateIcon);
 		this.add_child(box);
diff -Npur a/data/gnome-shell/pamac-updates@manjaro.org/metadata.json b/data/gnome-shell/pamac-updates@manjaro.org/metadata.json
--- a/data/gnome-shell/pamac-updates@manjaro.org/metadata.json	2026-10-08 08:54:42.906778117 -0600
+++ b/data/gnome-shell/pamac-updates@manjaro.org/metadata.json	2026-10-08 09:16:45.041147525 -0600
@@ -1,7 +1,7 @@
 {
   "description": "GNOME Shell Updates indicator for Pamac.",
   "name": "Pamac Updates Indicator",
-  "shell-version": ["45", "46", "47", "48", "49", "50"],
+  "shell-version": ["45", "46", "47", "48", "49", "50", "51"],
   "url": "https://github.com/manjaro/pamac",
   "uuid": "pamac-updates@manjaro.org",
   "gettext-domain": "pamac",
