diff --git a/panels/background/cc-background-panel.c b/panels/background/cc-background-panel.c
index 4cde2d0..688bff8 100644
--- a/panels/background/cc-background-panel.c
+++ b/panels/background/cc-background-panel.c
@@ -180,6 +180,8 @@ get_color_tooltip (GDesktopAccentColor color)
         return _("Purple");
     case G_DESKTOP_ACCENT_COLOR_SLATE:
         return _("Slate");
+    case G_DESKTOP_ACCENT_COLOR_MAIA:
+        return _("Manjaro Maia");
     default:
         g_assert_not_reached ();
     }
@@ -207,6 +209,8 @@ get_untranslated_color (GDesktopAccentColor color)
         return "purple";
     case G_DESKTOP_ACCENT_COLOR_SLATE:
         return "slate";
+    case G_DESKTOP_ACCENT_COLOR_MAIA:
+        return "maia";
     default:
         g_assert_not_reached ();
     }
@@ -218,7 +222,7 @@ setup_accent_color_toggles (CcBackgroundPanel *self)
     GDesktopAccentColor accent_color = g_settings_get_enum (self->interface_settings, INTERFACE_ACCENT_COLOR_KEY);
     GDesktopAccentColor i;
 
-    for (i = G_DESKTOP_ACCENT_COLOR_BLUE; i <= G_DESKTOP_ACCENT_COLOR_SLATE; i++) {
+    for (i = G_DESKTOP_ACCENT_COLOR_BLUE; i <= G_DESKTOP_ACCENT_COLOR_SLATE || i == G_DESKTOP_ACCENT_COLOR_MAIA; i++) {
         GtkWidget *button = GTK_WIDGET (gtk_toggle_button_new ());
         GtkToggleButton *grouping_button = GTK_TOGGLE_BUTTON (gtk_widget_get_first_child (self->accent_box));
 
@@ -235,6 +239,9 @@ setup_accent_color_toggles (CcBackgroundPanel *self)
             gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (button), TRUE);
 
         gtk_box_append (GTK_BOX (self->accent_box), button);
+
+        if (i == G_DESKTOP_ACCENT_COLOR_SLATE)
+            i = G_DESKTOP_ACCENT_COLOR_MAIA-1;
     }
 }
 
diff --git a/shell/style.css b/shell/style.css
index 3cbcbd4..780a230 100644
--- a/shell/style.css
+++ b/shell/style.css
@@ -165,7 +165,7 @@ flowboxchild.active-item .selected-check {
               0 0 0 6px color-mix(in srgb, var(--accent-color) 30%, transparent);
 }
 
-.blue, .teal, .green, .yellow, .orange, .red, .pink, .purple, .slate {
+.blue, .teal, .green, .yellow, .orange, .red, .pink, .purple, .slate, .maia {
   --accent-color: oklab(from var(--accent-bg-color) var(--standalone-color-oklab));
 }
 
@@ -324,7 +324,7 @@ flowboxchild.active-item .selected-check {
               0 0 0 6px color-mix(in srgb, var(--accent-color) 30%, transparent);
 }
 
-.blue, .teal, .green, .yellow, .orange, .red, .pink, .purple, .slate {
+.blue, .teal, .green, .yellow, .orange, .red, .pink, .purple, .slate, .maia {
   --accent-color: oklab(from var(--accent-bg-color) var(--standalone-color-oklab));
 }
 
@@ -364,6 +364,10 @@ flowboxchild.active-item .selected-check {
   --accent-bg-color: var(--accent-slate);
 }
 
+.maia {
+  --accent-bg-color: var(--accent-maia);
+}
+
 /* Color Calibration */
 .cc-color-calibrate-window {
   background-color: black;
