--- a/drivers/iio/imu/bmi270/bmi270_i2c.c	2026-07-30 16:16:29.446553201 +0200
+++ b/drivers/iio/imu/bmi270/bmi270_i2c.c	2026-07-30 16:22:22.710076622 +0200
@@ -41,8 +41,11 @@ MODULE_DEVICE_TABLE(i2c, bmi270_i2c_id);
 static const struct acpi_device_id bmi270_acpi_match[] = {
 	/* GPD Win Mini, Aya Neo AIR Pro, OXP Mini Pro, etc. */
 	{ "BMI0160",  (kernel_ulong_t)&bmi260_chip_info },
-	/* GPD Win Max 2 2023(sincice BIOS v0.40), etc. */
+	{ "BOSC0160", (kernel_ulong_t)&bmi260_chip_info },
+	/* GPD Win Max 2 2023(since BIOS v0.40), etc. */
 	{ "BMI0260",  (kernel_ulong_t)&bmi260_chip_info },
+	/* OrangePi Neo 2026 (since BIOS BCPH01_V1.20), etc. */
+	{ "BOSC0260", (kernel_ulong_t)&bmi260_chip_info },
 	{ }
 };
 MODULE_DEVICE_TABLE(acpi, bmi270_acpi_match);
