diff -Npur a/arkdep b/arkdep
--- a/arkdep	2026-06-10 08:25:12.406283730 -0600
+++ b/arkdep	2026-06-10 08:28:01.701693791 -0600
@@ -13,8 +13,8 @@ set -o pipefail
 # setting is not configured and as the default settings a new system will
 # be initialized with.
 declare -r dist_enable_overlay=1
-declare -r dist_repo_url='https://repo.arkanelinux.org/arkdep'
-declare -r dist_repo_default_image='arkanelinux'
+declare -r dist_repo_url='https://download.manjaro.org/arkdep'
+declare -r dist_repo_default_image='manjaro-summit-gnome'
 declare -r dist_deploy_keep=3
 declare -r dist_clean_cache_on_remove=1
 declare -r dist_always_healthcheck=1
@@ -23,7 +23,7 @@ declare -r dist_minimum_available_boot_s
 declare -r dist_minimum_available_root_storage=12582912
 declare -r dist_update_cpu_microcode=1
 declare -r dist_backup_user_accounts=0
-declare -r dist_migrate_files="'var/usrlocal' 'var/opt' 'var/srv' 'var/lib/AccountsService' 'var/lib/bluetooth' 'var/lib/NetworkManager' 'var/lib/arkane' 'var/lib/power-profiles-daemon' 'var/db' 'etc/localtime' 'etc/locale.gen' 'etc/locale.conf' 'etc/NetworkManager/system-connections' 'etc/ssh' 'etc/fstab' 'etc/crypttab' 'etc/luks-keys' 'etc/passwd' 'etc/shadow' 'etc/group' 'etc/subuid' 'etc/subgid'"
+declare -r dist_migrate_files="'var/usrlocal' 'var/opt' 'var/srv' 'var/lib/AccountsService' 'var/lib/bluetooth' 'var/lib/NetworkManager' 'var/lib/arkane' 'var/lib/power-profiles-daemon' 'var/db' 'etc/localtime' 'etc/locale.gen' 'etc/locale.conf' 'etc/NetworkManager/system-connections' 'etc/ssh' 'etc/fstab' 'etc/crypttab' 'etc/luks-keys' 'etc/passwd' 'etc/shadow' 'etc/group' 'etc/subuid' 'etc/subgid' 'var/lib/manjaro-branch'"
 declare -r dist_load_extensions=0
 declare -r dist_remove_tar_after_deployment=1
 declare -r dist_update_diff_style='list'
@@ -34,8 +34,8 @@ declare -r dist_auto_update_templates=1
 
 # Default configuration file settings used for init
 # systemd-boot configuration, this is just the bare minimum needed for the system to function
-declare -r boot_entry_title='Arkane Linux'
-declare -r kernel_params='root="LABEL=arkane_root" rootflags=subvol=/arkdep/deployments/%target%/rootfs rw'
+declare -r boot_entry_title='Manjaro Linux - Arkdep'
+declare -r kernel_params='root="LABEL=manjaro_root" rootflags=subvol=/arkdep/deployments/%target%/rootfs iomem=relaxed quiet splash rw'
 
 ## Check if program is locked, if yes do not run
 #
