diff -Npur a/util/grub.d/30_uefi-firmware.in b/util/grub.d/30_uefi-firmware.in
--- a/util/grub.d/30_uefi-firmware.in	2026-10-07 07:22:41.024364423 -0600
+++ b/util/grub.d/30_uefi-firmware.in	2026-10-07 07:25:26.827246466 -0600
@@ -33,11 +33,8 @@ gettext_printf "Adding boot menu entry f
 allowed_user_list="$(grub_get_auth_user_list $GRUB_RECOVERY_ACCESS_LEVEL)"
 cat << EOF
 if [ "\$grub_platform" = "efi" ]; then
-	fwsetup --is-supported
-	if [ "\$?" = 0 ]; then
-		menuentry '$LABEL' \$menuentry_id_option 'uefi-firmware' ${allowed_user_list}{
+	menuentry '$LABEL' \$menuentry_id_option 'uefi-firmware' ${allowed_user_list}{
 			fwsetup
 		}
-	fi
 fi
 EOF
